examples/benchmarks.py

```diff
@@ -250,15 +257,21 @@
 
 def create_setup_and_compute(
```

Comment:
I definitely lack context, but I feel like I would want to use this with 1 model, 1 batch_size, one  slice size. The fact that the code is taking lists of models, lists of batch sizes, and lists of slice sizes adds a fair amount of complexity, (e.g. `results[model_name]["memory"][batch_size][slice_size]` would just be `results['memory']`. 

If the use case is comparing different runs I guess the signature makes sense. Is that the reasoning?

Reply:
Yes, that the script we used for this blog post for instance: https://medium.com/huggingface/benchmarking-transformers-pytorch-and-tensorflow-e2917fb891c2

I kept it as is, just added memory benchmarking in addition to speed (and a little more flexibility in the CL args)